fe/fe-core/src/main/java/org/apache/doris/catalog/ExternalCatalogResource.java

```diff
@@ -0,0 +1,108 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.catalog;
+
+import org.apache.doris.common.DdlException;
+import org.apache.doris.common.proc.BaseProcResult;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.Map;
+
+/**
+ * External Catalog resource for external table query.
+ *
+ * External Catalog resource example:
+ * CREATE EXTERNAL RESOURCE "odbc_mysql"
+ * PROPERTIES
+ * (
+ *     "type" = "external_catalog", [required]
```

Comment:
The current hierarchy is:
Resource -> external_catalog -> odbc_catalog

But I think it can be simplified as:
Resource -> odbc_catalog

So the "type" is "odbc_catalog", and "odbc_type" is "mysql".